Apply pre-4.4.0 release changes (#1674)
mongodb/mongo-cxx-driver: etc/releasing.md

See the [MongoDB C++ Driver Manual](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/) and the [Driver Installation Instructions](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/installation/) for more details on downloading, installing, and using this driver.

> NOTE: The mongocxx X.Y.x series does not promise ABI stability of entities in the `v_noabi` namespace.

Reviewer: ```suggestion
NOTE: ABI stability is NOT supported for entities declared in the `v_noabi` namespace.
```

Wording tweak: stronger language for clarity.

Confirming: removal of the old note suggests we are committing to _not_ having [an escape hatch](https://github.com/mongodb/mongo-c-driver/pull/2312#discussion_r3350088426) for making API/ABI breaking changes in a patch release. This is consistent with the C++ Driver's current [Versioning policy](https://www.mongodb.com/docs/languages/cpp/cpp-driver/current/reference/api-abi-versioning/), but I'd overlooked the note in this document.

Author: Updated as suggested.

> Confirming: removal of the old note suggests we are committing to _not_ having [an escape hatch](https://github.com/mongodb/mongo-c-driver/pull/2312#discussion_r3350088426)

I did not intend to commit to no escape hatch. I expect this note was a relic of old practices. The C++ driver used to have an (unfortunate) practice of deprecating API in a minor release, and removing in a later minor release. Example: CXX-1578.
